release/models/bgp/openconfig-bgp-common.yang

```diff
@@ -698,17 +704,52 @@ submodule openconfig-bgp-common {
     description
       "Parameters controlling usage of  of DMZ Link-Bandwidth
       extended community in pultipath RIB/FIB formation";
+
     leaf enabled {
       type boolean;
       description
-      "When set to TRUE, BGP multiplepath shall distributed traffic
-      load among contributing routes proportionally to value of
-      Local Administrator subfield of link-bandwidth extended
-      community [draft-ietf-idr-link-bandwidth-07].
-      This leaf has no effect if BGP multi-path is disabled or
-      if maximum-path attribute of BGP multi-path value is set
-      to 1";
+        "When set to TRUE, BGP multiplepath shall distributed traffic
+        load among contributing routes proportionally to value of
+        Local Administrator subfield of link-bandwidth extended
+        community [draft-ietf-idr-link-bandwidth-07].
+        This leaf has no effect if BGP multi-path is disabled or
+        if maximum-path attribute of BGP multi-path value is set
+        to 1";
+    }
+
+    leaf cumulative {
+      type boolean;
+      when "../enabled = 'TRUE'";
```

Comment:
The "enabled" above is "do load balancing locally".  If the device is not locally performing ECMP but is a pinch-point for routes and thus sending the aggregated link-bw, advertising the aggregate may still be appropriate.  Perhaps consider removing this constraint.

Reply:
As I commented, IMHO this is anchored in wrong place. The "non-transitive-send-over-eBGP" is orthogonal to multipath.

Reply:
Very broadly speaking, as I've discussed internally with Juniper developers, there's some ugly ambiguity in RFC 4360 with regard to non-transitive extended communities.  The ambiguity comes from unclear semantics for propagation vs. "origination".

- If the community is unmolested locally, is learned from another iBGP speaker, then it should be dropped at the AS boundary.
- If the community is attached as part of export procedures to an eBGP speaker, the community is effectively "originated" at that boundary.  A valid semantic would be that AS1 pushes the community into AS2 with the expectation that it doesn't leave AS2.

This second sense isn't in the RFC, but overlaps the point here for the DMZ draft.  

In those senses, I agree that this is a generic problem.  There's some motivation to do a RFC 4360-bis to clarify the point, but that work hasn't happened.

Meanwhile, implementations have inconsistent knobs for dealing with this scenario.  Right now, the only compelling use case (non-transitive ext-comms are rare) is this DMZ case.  Thus, having a knob here is "fine".  Dealing with the broader discussion is likely outside the OC use case until implemented by enough vendors.

Reply:
Added `non-transitive` boolean leaf to explicitly indicate if link-bandwidth communities from iBGP peers will be sent to eBGP peers.  WDYT?